presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/ResourceGroupId.java

```diff
@@ -71,6 +78,15 @@ public String getLastSegment()
         return Optional.of(new ResourceGroupId(segments.subList(0, segments.size() - 1)));
     }
 
+    public boolean isAncestor(ResourceGroupId other)
```

Comment:
`isAncestor` is ambiguous as to whether:
* `other` is ancestor of `this`; or
* `this` is ancestor of `other`

We need to do two things to make it clear
* rename the method to `isAncestorOf`; and
* rename `other` and `otherSegments` to `descendant` and `descendantSegments`.